src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -241,9 +345,143 @@ def visit_doctor(
 
         return df
 
-    def update_treatment(self, index: pd.Index):
-        # TODO [MIC-3371, MIC-3375]
-        pass
+    def update_treatment(self, df: pd.DataFrame) -> Tuple[pd.DataFrame, pd.Index]:
+        """Applies treatment ramps
+        
+        Arguments:
+            df: state table subset to include only simulants who visit the doctor
+        """
+        df["to_schedule"] = np.nan  # temporary column
+        df = self.apply_sbp_treatment_ramp(df)
+        df = self.apply_ldlc_treatment_ramp(df)
+        return df, df[df["to_schedule"].notna()].index
+
+    def treat_not_currently_medicated_sbp(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Applies the SBP treatment ramp to simulants not already on medication
+        
+        Arguments:
+            df: dataframe of simulants subset to those visiting the doctor and
+                not already medicated
+        """
+
+        # Generate indexes for simulants who overcome therapeutic inertia and
+        # who have medium and high (measured) SBP levels
+        overcome_therapeutic_inertia = df.index[
+            self.randomness.get_draw(df.index) > data_values.THERAPEUTIC_INERTIA_NO_START
+        ]
+        mid_sbp = df[
+            (df["measured_level"] >= data_values.SBP_THRESHOLD.LOW)
+            & (df["measured_level"] < data_values.SBP_THRESHOLD.HIGH)
+        ].index
+        high_sbp = df[df["measured_level"] >= data_values.SBP_THRESHOLD.HIGH].index
+
+        # Low-SBP patients: do nothing
+
+        # Mid-SBP patients
+        to_prescribe_mid_sbp = mid_sbp.intersection(overcome_therapeutic_inertia)
+        do_not_prescribe_mid_sbp = mid_sbp.difference(to_prescribe_mid_sbp)
+        # If no tx prescribed, still schedule a followup
+        df.loc[do_not_prescribe_mid_sbp, "to_schedule"] = 1
+        # If tx prescribed, it must be one_drug_half_dose; also schedule followup
+        df.loc[
+            to_prescribe_mid_sbp, self.sbp_medication_column
+        ] = data_values.MEDICATION_RAMP["sbp"][
+            data_values.SBP_MEDICATION_LEVEL.ONE_DRUG_HALF_DOSE
+        ]
+        df.loc[to_prescribe_mid_sbp, "to_schedule"] = 1
```

Comment:
You can just say schedule all people with mid-sbp

Reply:
Ah, yeah that was leftover from when I commented out all decision points in the ramp tree